### PR TITLE
Quickmenu unlock position from head rotation and angle configuration

### PR DESCRIPTION
**Pull request by Aceship**
- Change Quickmenu to follow position of the player, but not the head rotation
- Add Quickmenu angle to configuration

### CODE REVIEW

**Comment by Aceship** [comment]
Add some Spear Throwing enhancement 

- Only sheath spear if you ungrab the button (allows you to throw from back of your head)
- Cancel throw if you don't move your hand much
- Can throw slower
- Less random throwing direction
- Probably more accurate throwing
- Fix Throwing on the movement direction while moving

File: ValheimVRMod/Scripts/QuickAbstract.cs [hunk 1]
```diff
@@ -32,10 +32,13 @@ public abstract class QuickAbstract : MonoBehaviour {
         private void OnEnable() {
             transform.SetParent(parent, false);
             transform.localPosition = Vector3.zero;
-            transform.localRotation = Quaternion.Euler(107, 0, -5);
+            transform.localRotation = Quaternion.Euler(VHVRConfig.getQuickMenuAngle(), 0, -5);
```
Comment (Andi1986) [inline ValheimVRMod/Scripts/QuickAbstract.cs:35]: lets rather remove this line + the config and instead add a transform.LookAt

Comment (Aceship) [reply ValheimVRMod/Scripts/QuickAbstract.cs]: Added both camera and hand option

File: ValheimVRMod/VRCore/VRPlayer.cs [hunk 2]
```diff
@@ -859,16 +859,26 @@ private void checkHandOverShoulder(bool isRightHand, Hand hand, SteamVR_Input_So
                 {
                     toggleShowHand = false;
                     hand.hapticAction.Execute(0, 0.2f, 100, 0.3f, inputSource);
-                    
-                    if (isRightHand && EquipScript.getLeft() == EquipType.Bow) {
+                    if(isRightHand && isHoldingItem(isRightHand) && (EquipScript.getRight() == EquipType.Spear || EquipScript.getRight() == EquipType.SpearChitin)) {
+
```
Comment (Andi1986) [inline ValheimVRMod/VRCore/VRPlayer.cs:863]: means we cannot holster spear anymor :(

Comment (Aceship) [reply ValheimVRMod/VRCore/VRPlayer.cs:863]: you can, its just changed to unpress part

File: ValheimVRMod/VRCore/VRPlayer.cs [hunk 3]
```diff
@@ -859,16 +859,26 @@ private void checkHandOverShoulder(bool isRightHand, Hand hand, SteamVR_Input_So
...
+                    } else if (isRightHand && EquipScript.getLeft() == EquipType.Bow) {
                         BowLocalManager.instance.toggleArrow();
                     } else if (isHoldingItem(isRightHand)) {
                         getPlayerCharacter().HideHandItems();
                     } else {
                         getPlayerCharacter().ShowHandItems();
+                        justUnsheathed = true;
+                    }
+                }
+                else if (!justUnsheathed && isRightHand && action.GetStateUp(inputSource)) {
```
Comment (Aceship) [inline ValheimVRMod/VRCore/VRPlayer.cs:873]: @Andi1986 on this part

File: ValheimVRMod/VRCore/VRPlayer.cs [hunk 4]
```diff
@@ -859,16 +859,26 @@ private void checkHandOverShoulder(bool isRightHand, Hand hand, SteamVR_Input_So
...
+                    if (isHoldingItem(isRightHand) && (EquipScript.getRight() == EquipType.Spear || EquipScript.getRight() == EquipType.SpearChitin)) {
+                        getPlayerCharacter().HideHandItems();
                     }
                 }
             }
+            if (justUnsheathed && isRightHand && action.GetStateUp(inputSource)&& (EquipScript.getRight() == EquipType.Spear || EquipScript.getRight() == EquipType.SpearChitin)) {
```
Comment (Andi1986) [inline ValheimVRMod/VRCore/VRPlayer.cs:879]: it almost feels if these conditions could be optimized somehow

**Review by Andi1986 — APPROVED** [review]

File: ValheimVRMod/VRCore/VRPlayer.cs (hunk 4 above)
Comment (Aceship) [reply ValheimVRMod/VRCore/VRPlayer.cs:879]: yeah, but currently the only way i thought is that way

**Review by brandonmousseau — APPROVED** [review]

**Pull request merged**